src/components/SelectionListWithModal/index.tsx

```diff
@@ -26,22 +27,41 @@ function SelectionListWithModal<TItem extends ListItem>(
     // We need to use isSmallScreenWidth instead of shouldUseNarrowLayout here because there is a race condition that causes shouldUseNarrowLayout to change indefinitely in this component
     // See https://github.com/Expensify/App/issues/48675 for more details
     const {isSmallScreenWidth} = useResponsiveLayout();
-    const {selectionMode} = useMobileSelectionMode(true);
+    const {selectionMode} = useMobileSelectionMode(shouldAutoTurnOff);
+    const wasSelectionOnRef = useRef(false);
+    const selectionRef = useRef(0);
 
     useEffect(() => {
         // We can access 0 index safely as we are not displaying multiple sections in table view
-        const selectedItems = sections[0].data.filter((item) => item.isSelected);
+        const selectedItems = sections[0].data.filter((item) => !!item.isSelected || (item as unknown as ReportListItemType)?.transactions?.some((transaction) => transaction.isSelected));
```

Comment:
I don't think doing these castings is the right way of coding this. This component is implemented to be generic and here we are adding some very specific code in case `ListItem` happens to come from a `ReportListItemType`.

I think these list items with transactions are created here: 

https://github.com/Expensify/App/blob/df17d79223b1c0880277a22c89a7b26ac949bbdb/src/components/Search/index.tsx#L73

Isn't it a better solution to change that line 

```diff
-              isSelected: item.transactions.every((transaction) => selectedTransactions[transaction.keyForList]?.isSelected && canSelectMultiple),
+              isSelected: item.transactions.some((transaction) => selectedTransactions[transaction.keyForList]?.isSelected,
```

so we can again trust `item.isSelected` to have the right value?

cc @bernhardoj since you worked on this here: https://github.com/Expensify/App/pull/47034/files
Do you see any problem with the change done in this PR or the change I'm proposing above?

If the change I'm proposing above creates a bug, then my second suggestion would be to add a prop to `SelectionListWithModal` like:

`isSelected: (item: TItem) => bool` that we can override. Then we pass this prop where you want to do this special `isSelected` check considering transactions.

Reply:
I agree with this.

> This component SelectionListWithModal was implemented to be generic and here we are adding some very specific code in case ListItem happens to come from a ReportListItemType.

But not with this one since it will change the current behavior, unless we are fine with it.

> Isn't it a better solution to change that line
>```diff
>-              isSelected: item.transactions.every((transaction) => >selectedTransactions[transaction.keyForList]?.isSelected && canSelectMultiple),
>+              isSelected: item.transactions.some((transaction) => >selectedTransactions[transaction.keyForList]?.isSelected,
>```

Before:

Report checkbox is unselected if at least one transaction is unselected.

<img width="382" alt="image" src="https://github.com/user-attachments/assets/9cf58e7b-2951-4f6a-9a5f-335a3e06e5d5">


After:

Report checkbox is selected if at least one transaction is selected.

<img width="413" alt="image" src="https://github.com/user-attachments/assets/f52b440e-117a-45f3-bb63-6ab8cbb8a478">

I think passing the selected state as prop is a good idea.

Reply:
Thanks @bernhardoj !